opensrp-reveal/src/main/java/org/smartregister/reveal/adapter/TaskRegisterAdapter.java

```diff
@@ -66,6 +66,13 @@ public void onBindViewHolder(@NonNull TaskRegisterViewHolder viewHolder, int pos
             name = context.getString(R.string.bcc);
             action = context.getString(R.string.record_bcc);
             hasIcon = true;
+        } else {
+            if (name == null) {
+                name = task.getFamilyName() != null ? task.getFamilyName() : "Structure " + random.nextInt(100);
```

Comment:
Is the assumption here that there will be few structures without names?

Since there would be name collisions otherwise.

Reply:
For structures without Name they are being displayed as **_Unenumerated structure_**